be/src/exprs/arithmetic_operation.h

```diff
@@ -281,11 +281,13 @@ struct ArithmeticBinaryOperator {
                 // return 0 when mod by -0.
                 // adjusted_r = r == 0.0 ? r + 1.0 : r;
                 r_is_zero = b.CreateFCmpOEQ(r, llvm::ConstantFP::get(r->getType(), 0));
-                auto* r_is_negative_zero = b.CreateFCmpOEQ(r, llvm::ConstantFP::get(r->getType(), -0));
+                auto* r_is_negative_zero = b.CreateFCmpOEQ(r, llvm::ConstantFP::get(r->getType(), -0.0));
                 auto* sum = b.CreateFAdd(r, llvm::ConstantFP::get(r->getType(), 1));
                 auto* adjusted_r = b.CreateSelect(r_is_zero, sum, r);
                 result.value = b.CreateSelect(r_is_negative_zero, llvm::ConstantFP::get(r->getType(), 0),
                                               b.CreateFRem(l, adjusted_r));
+                auto* result_is_nagative_zero = b.CreateFCmpOEQ(result.value, llvm::ConstantFP::get(result.value->getType(), -0.0));
+                result.value = b.CreateSelect(result_is_nagative_zero, llvm::ConstantFP::get(result.value->getType(), 0.0), result.value);
```

Comment:
what about interger types?

Reply:
The case of int is not considered in the original code, probably because int does not have -0?
``` cpp
else if constexpr (is_mod_op<Op>) {
            if constexpr (lt_is_float<Type>) {
                auto result = fmod(l, (r + (r == 0)));
                if (UNLIKELY(result == -0)) {
                    return 0;
                }
                return result;
            } else if constexpr (may_cause_fpe<ResultType>) {
                if (UNLIKELY(check_fpe_of_min_div_by_minus_one(l, r))) {
                    return 0;
                } else {
                    return l % (r + (r == 0));
                }
            } else {
                return l % (r + (r == 0));
            }
        }
```